### PR TITLE
[SQL-125] Redact sensitive information from traces

### PR DESCRIPTION
**Pull request by kelvinqian00**
- Redact sensitive info before logging config var maps
- ~Turn off Pedestal logging since we can't redact info from that~ (Made redundant by https://github.com/yetanalytics/lrs/pull/71)
- Return 400 instead of 500 error (which spews out exn info) upon JSONParseException
- ~TODO: Write tests for the above~ Done!

### CODE REVIEW

File: src/main/lrsql/system/database.clj [hunk 1]
```diff
@@ -97,7 +97,7 @@
       (if-not ?conn-pool
         (let [conn-pool (make-conn-pool backend config)]
           (log/infof "Starting new connection for %s database..." db-type)
-          (log/debugf "Config: %s" config)
+          (log/debugf "Config: %s" (cu/redact-config-vars config))
```
Comment (milt) [inline src/main/lrsql/system/database.clj:100]: After reading the `c.t.l` source, the walk is fine. If this only happens in debug, the expanded macro contains a conditional on level.

**Review by milt — APPROVED** [review]
Looks sensible enough to me

**Review by cliffcaseyyet — APPROVED** [review]
nice

**Review by milt — APPROVED** [review]
Yep, looks good. As expected the json parse catcher after the general error handler will get called before it on the exit interceptor pass.

**Review by cliffcaseyyet — APPROVED** [review]

**Pull request merged**